### PR TITLE
Fix from attribute

### PR DESCRIPTION
**Pull request by SimonDamberg**
- `TelegramObject.__getitem__` now raises `KeyError` instead of returning `None`, and a special case is made for `obj['from']` to look up `obj['from_user']`. 
- Removed `__getitem__` override from `Message`
- Test for Message changed to check for `KeyError` instead of `None`.

### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Added myself alphabetically to `AUTHORS.rst` (optional)

### CODE REVIEW

**Comment by Bibo-Joshi** [comment]
> There's a pre-commit test failing. Can you add a pylint ignore on that line? see the other places in the code on how to add a pylint ignore.

rather just do what pylint suggests to do ;)

**Comment by Bibo-Joshi** [comment]
@SimonDamberg are you still interested in finishing the PR? :)

**Comment by SimonDamberg** [comment]
A lot of things came up at school and job. If someone wants to clean up and finish, feel free to do so :)

File: telegram/_telegramobject.py [hunk 1]
```diff
@@ -34,7 +34,16 @@
 
 
 class TelegramObject:
-    """Base class for most Telegram objects."""
+    """Base class for most Telegram objects.
+
+    Objects of this type a subscriptable with strings, where ``telegram_object[attribute_name]`` is
```
Comment (harshil21) [inline telegram/_telegramobject.py:39]:
```suggestion
    Objects of this type are subscriptable with strings, where ``telegram_object[attribute_name]`` is
```

File: telegram/_telegramobject.py [hunk 2]
```diff
@@ -34,7 +34,16 @@
...
+    equivalent to ``telegram_object.attribute_name``. If the object does not have an attribute with
+    the appropriate name, a :class:`KeyError` will be raised.
```
Comment (harshil21) [inline telegram/_telegramobject.py:41]:
```suggestion
    the appropriate name, a :exc:`KeyError` will be raised.
```
I think this plays in better with intersphinx

**Review by harshil21 — CHANGES_REQUESTED** [review]

**Review by harshil21 — APPROVED** [review]

**Comment by Bibo-Joshi** [comment]
Thanks for getting this started @SimonDamberg :)

**Pull request merged**